File: backend/models/group.go
package models

import "time"

// Group 小组模型
type Group struct {
	ID        uint      `json:"id" gorm:"primaryKey"`
	Name      string    `json:"name" gorm:"not null"`
	PMID      uint      `json:"pm_id" gorm:"not null;index"`
	PM        User      `json:"pm,omitempty" gorm:"foreignKey:PMID"`
	DevLeadID uint      `json:"dev_lead_id" gorm:"not null;index"`
	DevLead   User      `json:"dev_lead,omitempty" gorm:"foreignKey:DevLeadID"`
	CreatedAt time.Time `json:"created_at"`
}

File: backend-java/src/main/java/com/management/group/entity/Group.java
package com.management.group.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.management.user.entity.User;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("`groups`")
public class Group {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private Long pmId;
    @TableField(exist = false)
    private User pm;
    private Long devLeadId;
    @TableField(exist = false)
    private User devLead;
    private String leadRole;
    private LocalDateTime createdAt;
}
